das/metrics.go

```diff
@@ -15,6 +15,12 @@ import (
 	"github.com/celestiaorg/celestia-node/header"
 )
 
+const (
+	jobTypeLabel     = "job_type"
+	headerWidthLabel = "header_width"
+	failedLabel      = "failed"
+)
```

Comment:
why did you choose to go with this approach? I remember trying this before and there was feedback and preference to not use constants as the compiler will strip them away anyway. I personally am in favor of them generally, they make things easier to update

Reply:
Why do you think compiler will strip them away? As far as I understand constants should be more efficient, because they are not allocated in the runtime.
But main motivation is the fact that labels are reused for multiple metrics and I wanted to be sure they are aligned nicely everywhere.

Reply:
@derrandz IIRC that feedback was for single use strings that were extracted as consts (which are automatically set by compiler) - if there are multiple uses its fine